compact/test/Compiler.test.ts

```diff
@@ -0,0 +1,686 @@
+        '--skip-zk',
+        'security',
+        '0.24.0',
+        mockExec,
+      );
+
+      expect(compiler).toBeInstanceOf(CompactCompiler);
+    });
+
+    it('should trim flags', () => {
+      compiler = new CompactCompiler('  --skip-zk --verbose  ');
+      expect(compiler.testFlags).toBe('--skip-zk --verbose');
+    });
+  });
+
+  describe('fromArgs', () => {
+    it('should parse empty arguments', () => {
+      compiler = CompactCompiler.fromArgs([]);
+
+      expect(compiler.testFlags).toBe('');
+      expect(compiler.testTargetDir).toBeUndefined();
+      expect(compiler.testVersion).toBeUndefined();
+    });
+
+    it('should handle SKIP_ZK environment variable', () => {
+      compiler = CompactCompiler.fromArgs([], { SKIP_ZK: 'true' });
+
+      expect(compiler.testFlags).toBe('--skip-zk');
+    });
+
+    it('should ignore SKIP_ZK when not "true"', () => {
+      compiler = CompactCompiler.fromArgs([], { SKIP_ZK: 'false' });
+
+      expect(compiler.testFlags).toBe('');
+    });
+
+    it('should parse --dir flag', () => {
+      compiler = CompactCompiler.fromArgs(['--dir', 'security']);
+
+      expect(compiler.testTargetDir).toBe('security');
+      expect(compiler.testFlags).toBe('');
+    });
+
+    it('should parse --dir flag with additional flags', () => {
+      compiler = CompactCompiler.fromArgs([
+        '--dir',
+        'security',
+        '--skip-zk',
+        '--verbose',
+      ]);
+
+      expect(compiler.testTargetDir).toBe('security');
+      expect(compiler.testFlags).toBe('--skip-zk --verbose');
+    });
+
+    it('should parse version flag', () => {
+      compiler = CompactCompiler.fromArgs(['+0.24.0']);
+
+      expect(compiler.testVersion).toBe('0.24.0');
+      expect(compiler.testFlags).toBe('');
+    });
+
+    it('should parse complex arguments', () => {
+      compiler = CompactCompiler.fromArgs([
+        '--dir',
+        'security',
+        '--skip-zk',
+        '--verbose',
+        '+0.24.0',
+      ]);
+
+      expect(compiler.testTargetDir).toBe('security');
+      expect(compiler.testFlags).toBe('--skip-zk --verbose');
+      expect(compiler.testVersion).toBe('0.24.0');
+    });
+
+    it('should combine environment variables with CLI flags', () => {
+      compiler = CompactCompiler.fromArgs(['--dir', 'access', '--verbose'], {
+        SKIP_ZK: 'true',
+      });
+
+      expect(compiler.testTargetDir).toBe('access');
+      expect(compiler.testFlags).toBe('--skip-zk --verbose');
+    });
+
+    it('should throw error for --dir without argument', () => {
+      expect(() => CompactCompiler.fromArgs(['--dir'])).toThrow(
+        '--dir flag requires a directory name',
+      );
+    });
+
+    it('should throw error for --dir followed by another flag', () => {
+      expect(() => CompactCompiler.fromArgs(['--dir', '--skip-zk'])).toThrow(
+        '--dir flag requires a directory name',
+      );
+    });
+  });
+
+  describe('validateEnvironment', () => {
```

Comment:
We should add a test that validates an Error is thrown when an invalid Compact version is passed